opacus/tests/grad_samples/linear_test.py

```diff
@@ -27,15 +28,16 @@ def test_input_bias(
         H: int,
         input_dim: int,
         bias: bool,
+        batch_first: bool,
     ):
 
-        if input_dim == 2:
-            x_shape = [N, W]
         if input_dim == 3:
             x_shape = [N, Z, W]
         if input_dim == 4:
             x_shape = [N, Z, H, W]
 
         linear = nn.Linear(W, W + 2, bias=bias)
         x = torch.randn(x_shape)
-        self.run_test(x, linear, batch_first=True)
+        if not batch_first:
+            x = x.transpose(0, 1)
```

Comment:
FYI: I also added the transpose in this test, even though the test passes it without it. The test passes regardless of this line, as the input is just bunch of random numbers and the 0'th and fist dimension do not matter in this test

Reply:
Ok, I see what you mean - for Linear and Embedding the first dimensions doesn't matter at all, so when we have `batch_first=False` and `x_shape = [N, Z, W]`, then batch dimension is `Z` not `N`. This is what confused me as a reader.